pkg/log/log.go

```diff
@@ -95,10 +383,59 @@ func GoroutineID() int {
 	return id
 }
 
-type goroutineHook struct{}
+// CallerHandler adds caller information to log records.
+type CallerHandler struct {
+	handler slog.Handler
+}
+
+func (h *CallerHandler) Enabled(ctx context.Context, level slog.Level) bool {
+	return h.handler.Enabled(ctx, level)
+}
+
+func (h *CallerHandler) Handle(ctx context.Context, record slog.Record) error {
+	// Add caller information
+	if pc, file, line, ok := runtime.Caller(5); ok { // Adjust stack depth as needed
```

Comment:
Why 5? Have we counted the frames between this call and the Msg or Msgf() call?

Reply:
That seems to be doing the right thing.

Reply:
I suggest we use a constant.
I see in https://github.com/rs/zerolog/blob/master/globals.go#L66 they have a constant (they also have logic to count the frames to skip, but that may be o overengineering)

Reply:
Fixed.